[Issue]: Incompatible with latest libfabric v2.0.0

### Problem Description

ofiwg libfabric removed the definition of `container_of` in rdma/fabric.h in v2.0.0. aws-ofi-rccl relies on it, so compiling fails:

```
nccl_ofi_net.c:1005:9: error: call to undeclared function 'container_of'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                req = container_of(cq_entry[comp_idx].op_context,
                      ^
```

aws-ofi-nccl needed a similar fix:
https://github.com/aws/aws-ofi-nccl/pull/605

### Operating System

N/A

### CPU

N/A

### GPU

N/A

### ROCm Version

ROCm 6.0.0

### ROCm Component

_No response_

### Steps to Reproduce

_No response_

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

_No response_

--- a/include/nccl_ofi.h
+++ b/include/nccl_ofi.h
@@ -36,6 +36,13 @@ extern "C" {
 #define MAX_PROV_INFO		(15)
 #define MAX_BDF_LEN		(25)
 
+/* Copied from libfabric:rdma/fabric.h@30ec628: "libfabric: Initial commit" */
+#ifndef container_of
+#define container_of(ptr, type, field) \
+	((type *) ((char *)ptr - offsetof(type, field)))
+#endif
+/* end of copied libfabric macros */
+
 /*
  * NCCL_NET_HANDLE_MAXSIZE is a limited resource (and defined in NCCL).
  * An endpoint address buffer of 56 bytes *should* be large enough to hold

